### DISCUSSION TITLE
can we Authenticate and Authorise the user in keycloak without password

### DISCUSSION THREAD

**Discussion by kumararvindibs** [keycloak/keycloak#19386] [Q&A]
I want to login into the keycloak using api by providing the only userName, not the password. It means logged in into the keycloak without password.
Is it possible? If yes then let me know by which api I can achive this.

I tired a way to get the credential from DB side with the userId but the Credentials response doesn't contain the value of the password.

Thanks in advance.

**Comment by itbill** [comment]
Try to play with the browser authentication flow. The behavior can be achieved by replacing the "Username and password form" step with "Username form".
